mysql-test/columnstore/basic/t/union.test

```diff
@@ -17,46 +17,279 @@ CREATE TABLE t1 (a INT, b INT UNSIGNED);
 SELECT * FROM (SELECT a FROM t1 UNION SELECT b FROM t1) tu ORDER BY a;
 SELECT * FROM (SELECT b FROM t1 UNION SELECT a FROM t1) tu ORDER BY b;
+INSERT INTO t1 VALUES (-1,1);
+SELECT * FROM (SELECT a FROM t1 UNION SELECT b FROM t1) tu ORDER BY a;
+SELECT * FROM (SELECT b FROM t1 UNION SELECT a FROM t1) tu ORDER BY b;
+INSERT INTO t1 VALUES (-2,2);
+SELECT * FROM (SELECT * FROM t1 UNION SELECT * FROM t1) tu;
+SELECT * FROM (SELECT a FROM t1 UNION SELECT b FROM t1) tu ORDER BY a;
+SELECT * FROM (SELECT b FROM t1 UNION SELECT a FROM t1) tu ORDER BY b;
 DROP TABLE t1;
 
+--echo #
+--echo # Same Int Type Union
+--echo #
+
+CREATE TABLE t1 (a INT UNSIGNED, b INT UNSIGNED);
+INSERT INTO t1 VALUES (1, 1), (1, 1), (2, 1), (2, 1), (3, 2), (4, 2), (5, 2);
+SELECT * FROM (SELECT a FROM t1 UNION SELECT b FROM t1) tu ORDER BY a;
+SELECT * FROM (SELECT b FROM t1 UNION SELECT a FROM t1) tu ORDER BY b;
+SELECT * FROM (SELECT * FROM t1 UNION SELECT * FROM t1) tu;
+DROP TABLE t1;
+
+CREATE TABLE t1 (a INT, b INT);
+INSERT INTO t1 VALUES (-1, -1), (-1, -1), (2, 1), (2, 1), (3, 2), (4, 2), (5, 2);
+SELECT * FROM (SELECT a FROM t1 UNION SELECT b FROM t1) tu ORDER BY a;
+SELECT * FROM (SELECT b FROM t1 UNION SELECT a FROM t1) tu ORDER BY b;
+SELECT * FROM (SELECT * FROM t1 UNION SELECT * FROM t1) tu;
+DROP TABLE t1;
+
+--echo #
+--echo # Signed Int Union
+--echo #
+
+CREATE TABLE t1 (a TINYINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFFFFFF80 AS SIGNED)+2),(-1),(0),(1),(0x7F);
+CREATE TABLE t2 (a SMALLINT);
+INSERT INTO t2 VALUES (CAST(0xFFFFFFFFFFFF8000 AS SIGNED)+2),(-1),(0),(1),(0x7FFF);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+CREATE TABLE t1 (a TINYINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFFFFFF80 AS SIGNED)+2),(-1),(0),(1),(0x7F);
+CREATE TABLE t2 (a MEDIUMINT);
+INSERT INTO t2 VALUES (CAST(0xFFFFFFFFFF800000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFF);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+CREATE TABLE t1 (a TINYINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFFFFFF80 AS SIGNED)+2),(-1),(0),(1),(0x7F);
+CREATE TABLE t2 (a INT);
+INSERT INTO t2 VALUES (CAST(0xFFFFFFFF80000000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFFFF);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+CREATE TABLE t1 (a TINYINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFFFFFF80 AS SIGNED)+2),(-1),(0),(1),(0x7F);
+CREATE TABLE t2 (a BIGINT);
+INSERT INTO t2 VALUES (CAST(0x8000000000000000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFFFFFFFFFFFF);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
```

Comment:
In the above 4 cases, the same t1 is dropped and inserted multiple times, you can create t1 just once and reuse.